### PR TITLE
Changes for CASSANDRA-12700: test for corrupt auth table

### PR DESCRIPTION
**Pull request by jeffjirsa**
To be merged if and only iff CASSANDRA-12700 is committed in its current form.

### CODE REVIEW

File: auth_test.py [hunk 1]
```diff
@@ -181,6 +181,35 @@ def list_users_test(self):
         self.assertFalse(users['cathy'])
         self.assertTrue(users['dave'])
 
+    @since('2.2')
+    def handle_corrupt_role_data_test(self):
+        """
```
Comment (ptnapoleon) [inline auth_test.py]: Could you add
`@jira_ticket CASSANDRA-12700` to this docstring?

File: auth_test.py [hunk 2]
```diff
@@ -181,6 +181,35 @@ def list_users_test(self):
...
+        * Confirm that corrupted user can no longer login
+        """
+        self.prepare()
+
+        session = self.get_session(user='cassandra', password='cassandra')
+        session.execute("CREATE USER bob WITH PASSWORD '12345' SUPERUSER")
+
+        session2 = self.get_session(user='bob', password='12345')
+        rows = list(session2.execute("LIST USERS"))
+        self.assertEqual(2, len(rows))
```
Comment (ptnapoleon) [inline auth_test.py]: Can you use `assert_length_equal` from `tools.assertions`?

File: auth_test.py [hunk 3]
```diff
@@ -181,6 +181,35 @@ def list_users_test(self):
...
+        * Manually corrupt / delete the is_superuser cell of that role
+        * Confirm that listing users shows an invalid request
+        * Confirm that corrupted user can no longer login
+        """
+        self.prepare()
+
+        session = self.get_session(user='cassandra', password='cassandra')
+        session.execute("CREATE USER bob WITH PASSWORD '12345' SUPERUSER")
+
+        session2 = self.get_session(user='bob', password='12345')
```
Comment (ptnapoleon) [inline auth_test.py]: It's convention to name the session after the user, here `bob`

**Comment by jeffjirsa** [comment]
CASSANDRA-12700 has been committed, and I've pushed de6c9d4 to address points raised in your review @ptnapoleon

**Pull request merged**